### PR TITLE
feat(ckb|example): Export `buildRgbppTransferTx` from rgbpp package and update L1 transfer example

### PR DESCRIPTION
**Pull request by duanyytop**
## Changes

- Export `buildRgbppTransferTx` from rgbpp package to generate CKB virtual TX and BTC tx to be signed
- Update L1 transfer example with the function `buildRgbppTransferTx`

## Function Signature
```TypeScript
const buildRgbppTransferTx = async ({
  ckb: {
    collector,
    xudtTypeArgs,
    rgbppLockArgsList,
    transferAmount,
    ckbFeeRate,
  },
  btc: {
    fromBtcAddress,
    toBtcAddress,
    btcDataSource,
    fromPubkey,
    feeRate
  },
  isMainnet,
}: RgbppTransferTxParams): Promise<RgbppTransferTxResult>
```
The `collector` and `btcDataSource` will be constructed in the `rgbpp-sdk-service` with some necessary variables from the environment.  The necessary request and response for the L1 transfer in the `rgbpp-sdk-service` are as follows: 
```TypeScript
interface RgbppTransferReq {
  // The transferred RGB++ xUDT type script args
  xudtTypeArgs: string;           
  // The rgbpp assets cell lock script args array whose data structure is: out_index | btc_tx_id
  rgbppLockArgsList: string[];   
  // The xUDT amount to be transferred
  transferAmount: Hex;     
  // The sender BTC address       
  fromBtcAddress: string;     
  // The receiver BTC address
  toBtcAddress: string;     
}

interface RgbppTransferResp {
  // The JSON string for the `BtcTransferVirtualTxResult`
  ckbVirtualTxResult: string;    
  // The BTC PSBT hex string which can be used to construct Bitcoin PSBT
  btcPsbtHex: Hex;           
}
``` 

## Related PR
- #218

### CODE REVIEW

File: packages/rgbpp/src/rgbpp/types.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  fromBtcAddress: string;
+  // The receiver BTC address
+  toBtcAddress: string;
+  btcDataSource: DataSource;
+  isMainnet: boolean;
+}
+
+export interface RgbppTransferTxResult {
+  ckbVirtualTxResult: BtcTransferVirtualTxResult;
+  btcTxHexToSign: Hex;
```
Comment (ShookLyngs) [inline packages/rgbpp/src/rgbpp/types.ts:23]: I think `btcTxHexToSign` should be renamed to `btcPsbtHexToSign` or, more simply, `btcPsbtHex`.

**Review by ShookLyngs — CHANGES_REQUESTED** [review]

File: packages/rgbpp/src/rgbpp/types.ts (hunk 1 above)
Comment (duanyytop) [reply packages/rgbpp/src/rgbpp/types.ts:23]: Make sense for me

Comment (Flouse) [reply packages/rgbpp/src/rgbpp/types.ts]: https://github.com/ckb-cell/rgbpp-sdk/pull/216/commits/8ccd765f4ee750f2a0ee73d84a3100cd73a2a77d

**Review by Flouse — DISMISSED** [review]

File: packages/rgbpp/src/rgbpp/types.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  // The collector that collects CKB live cells and transactions
+  collector: Collector;
+  // The transferred RGB++ xUDT type script args
+  xudtTypeArgs: Hex;
+  // The rgbpp assets cell lock script args array whose data structure is: out_index | bitcoin_tx_id
+  rgbppLockArgsList: Hex[];
+  // The XUDT amount to be transferred, if the noMergeOutputCells is true, the transferAmount will be ignored
+  transferAmount: bigint;
+  // The sender BTC address
+  fromBtcAddress: string;
```
Comment (ShookLyngs) [inline packages/rgbpp/src/rgbpp/types.ts:14]: For P2TR inputs, the `fromPubkey` prop is required in the `sendRgbppUtxos()` API. BTW, do you think the `feeRate` prop should be exposed, too? ref: https://github.com/ckb-cell/rgbpp-sdk/pull/223/files#diff-45996e7b4d331bb196c3c3af127807387136cfdc33ff6de8cc604b40221b2b46R41

File: packages/rgbpp/src/rgbpp/types.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  // The receiver BTC address
+  toBtcAddress: string;
+  btcDataSource: DataSource;
```
Comment (ShookLyngs) [inline packages/rgbpp/src/rgbpp/types.ts:17]: Random idea out of nowhere—this is not a change request, just a topic of discussion: how about separating the props of BTC/CKB for better readability and easier prop management? For example:
```ts
{
  ckb: {
    collector: ...,
    xudtTypeArgs: ...,
    ...
  },
  btc: {
    from: ...,
    fromPubkey: ...,
    to: ...,
    ...
  },
}
```

File: packages/rgbpp/src/rgbpp/types.ts (hunk 2 above)
Comment (duanyytop) [reply packages/rgbpp/src/rgbpp/types.ts:14]: I just want to keep the parameters simple,  and we can discuss about which parameters are needed.

Adding `fromPubkey` and `feeRate` to parameters is OK for me.

File: packages/rgbpp/src/rgbpp/types.ts (hunk 3 above)
Comment (duanyytop) [reply packages/rgbpp/src/rgbpp/types.ts:17]: Maybe it's a good idea.

File: packages/rgbpp/src/rgbpp/types.ts (hunk 2 above)
Comment (duanyytop) [reply packages/rgbpp/src/rgbpp/types.ts:19]: Updated in https://github.com/ckb-cell/rgbpp-sdk/pull/216/commits/7bd4e7f40236d7d804d4a4085e85a5b6ed38c0ad

File: packages/rgbpp/src/rgbpp/types.ts (hunk 3 above)
Comment (duanyytop) [reply packages/rgbpp/src/rgbpp/types.ts:22]: Updated in https://github.com/ckb-cell/rgbpp-sdk/pull/216/commits/7bd4e7f40236d7d804d4a4085e85a5b6ed38c0ad

**Review by ShookLyngs — DISMISSED** [review]

File: packages/rgbpp/src/rgbpp/types.ts [hunk 4]
```diff
@@ -10,11 +10,25 @@ export interface RgbppTransferTxParams {
   rgbppLockArgsList: Hex[];
   // The XUDT amount to be transferred, if the noMergeOutputCells is true, the transferAmount will be ignored
   transferAmount: bigint;
+  // The CKB transaction fee rate, default value is 1100
+  ckbFeeRate?: bigint;
+}
+
+export interface RgbppTransferBtc {
```
[the remark this answers is not in the document]
Comment (duanyytop) [reply packages/rgbpp/src/rgbpp/types.ts:17]: I don't have a clear preference, adding the suffix is ​​also OK for me.

[the remark this answers is not in the document]
Comment (duanyytop) [reply packages/rgbpp/src/rgbpp/types.ts]: Updated in https://github.com/ckb-cell/rgbpp-sdk/pull/216/commits/42699bdca287b371d7b9ea79ef82cf441675b0e4

File: packages/rgbpp/src/rgbpp/types.ts [hunk 5]
```diff
@@ -10,11 +10,25 @@ export interface RgbppTransferTxParams {
...
   // The sender BTC address
   fromBtcAddress: string;
```
[the remark this answers is not in the document]
Comment (duanyytop) [reply packages/rgbpp/src/rgbpp/types.ts]: Updated in https://github.com/ckb-cell/rgbpp-sdk/pull/216/commits/42699bdca287b371d7b9ea79ef82cf441675b0e4

**Review by ShookLyngs — DISMISSED** [review]

**Review by Flouse — APPROVED** [review]

**Comment by duanyytop** [comment]
I fixed the conflicts between this branch and develop branch and then it needs to be reviewed again.

File: examples/rgbpp/xudt/2-btc-transfer.ts [hunk 6]
```diff
@@ -13,35 +13,26 @@ interface RgbppTransferParams {
...
-    transferAmount,
+  const { ckbVirtualTxResult, btcPsbtHex } = await buildRgbppTransferTx({
+    ckb: {
+      collector,
+      xudtTypeArgs,
+      rgbppLockArgsList,
+      transferAmount,
+    },
+    btc: {
+      fromAddress: btcAccount.from,
```
Comment (ShookLyngs) [inline examples/rgbpp/xudt/2-btc-transfer.ts:24]: Please fill the `fromPubkey` prop in the examples as it is required if the `fromAddress` is of `AddressType.P2TR`. Note that `btcAccount.fromPubkey` is undefined if `btcAccount.addressType == AddressType.P2WPKH`. ref: https://github.com/ckb-cell/rgbpp-sdk/blob/develop/examples/rgbpp/shared/btc-account.ts#L27-L52
```suggestion
      fromAddress: btcAccount.from,
      fromPubkey: btcAccount.fromPubkey,
```

**Review by ShookLyngs — CHANGES_REQUESTED** [review]

File: examples/rgbpp/xudt/2-btc-transfer.ts (hunk 6 above)
Comment (duanyytop) [reply examples/rgbpp/xudt/2-btc-transfer.ts:24]: I added `fromPubkey` to the BTC parameters for two RGB++ transfer examples at https://github.com/ckb-cell/rgbpp-sdk/pull/216/commits/21304feb90d57466c670acbe4e2912fe2726b474

**Review by ShookLyngs — APPROVED** [review]

**Review by Flouse — APPROVED** [review]

**Pull request merged**